cli/runtime/config/discovery_sources_node.go

```diff
@@ -106,3 +150,15 @@ func getDiscoverySourceTypeAndName(discoverySource configapi.PluginDiscovery) (s
 	}
 	return "", ""
 }
+
+// Find the matching discovery source type and index from accepted discovery sources
+func findDiscoverySourceTypeAndIndexByWeakMatch(currentNodes []*yaml.Node) (string, int) {
```

Comment:
nit: could use a more specific variable name, e.g discoverySourceContentNodes, so it does not look like this function is used on any generic node tree

Reply:
updated the variable name as suggested